src/Editor.jl

```diff
@@ -59,12 +60,113 @@ function ntbook_motion_notify_event_cb(widget::Ptr,  eventptr::Ptr, user_data)
 end
 signal_connect(ntbook_motion_notify_event_cb,editor,"motion-notify-event",Cint, (Ptr{Gtk.GdkEvent},), false)
 
-function close_tab()
-    idx = get_current_page_idx(editor)
-    splice!(editor,idx)
-    set_current_page_idx(editor,max(idx-1,0))
+
+function close_tab(idx::Int)
+  splice!(editor,idx)
+  set_current_page_idx(editor,max(idx-1,0))
+end
+close_tab() = close_tab(get_current_page_idx(editor))
+
+function close_tab_cb(btn::Ptr, tab)
+    close_tab(tab_num(editor,tab))
+    return nothing
+end
+
+function close_other_tabs_cb(btn::Ptr,tab)
+    while Gtk.GAccessor.n_pages(editor) > 1
+        if get_tab(editor,1) == tab
+            splice!(editor,2)
+        else
+            splice!(editor,1)
+        end
+    end
+    return nothing
+end
+function close_tabs_right_cb(btn::Ptr,tab)
+    idx = tab_num(editor,tab)
+    while (Gtk.GAccessor.n_pages(editor) > idx)
+        close_tab(idx+1)
+    end
+    return nothing
+end
+function close_all_tabs(btn::Ptr,tab)
+    while (Gtk.GAccessor.n_pages(editor) > 0)
+        close_tab(1)
+    end
+    return nothing
+end
+
+function find_filename(model_ptr, path_ptr, iter_ptr, data_ptr)
+  model = convert(Gtk.GtkTreeStoreLeaf,model_ptr)
+  iter  = unsafe_load(iter_ptr)
+  path  = Gtk.GtkTreePath(path_ptr)
+  data  = unsafe_pointer_to_objref(data_ptr)
+  if model[iter,3] == data[1]
+    data[2][1] = true
+    data[2][2] = path
+    return Cint(1)
+  else
+    return Cint(0)
+  end
+
+end
+@guarded nothing function reveal_in_tree_view(btn::Ptr, tab)
+    data = [false,nothing]
+    foreach(GtkTreeModel(filespanel.list),find_filename,(tab.filename,data))
+    if data[1]
+        set_cursor_on_cell(filespanel.tree_view, data[2])
+    end
+    return nothing
 end
+function create_tab_menu(container, tab)
+  menu =  @GtkMenu() |>
+  (closeTabItem = @GtkMenuItem("Close Tab")) |>
+  (closeOthersTabsItem = @GtkMenuItem("Close Others Tabs")) |>
+  (closeTabsRight = @GtkMenuItem("Close Tabs to the Right ")) |>
+  (closeAllTabs = @GtkMenuItem("Close All Tabs")) |>
+  @GtkSeparatorMenuItem() |>
+  (revealInTreeItem = @GtkMenuItem("Reveal in Tree View"))
+
+  signal_connect(close_tab_cb, closeTabItem, "activate", Void,(),false,tab)
+  signal_connect(close_other_tabs_cb, closeOthersTabsItem, "activate", Void,(),false,tab)
+  signal_connect(close_tabs_right_cb, closeTabsRight, "activate", Void,(),false,tab)
+  signal_connect(close_all_tabs, closeAllTabs, "activate", Void,(),false,tab)
+  signal_connect(reveal_in_tree_view, revealInTreeItem, "activate", Void,(),false,tab)
+
+  showall(menu)
+  return menu
 
+end
+
+  function tab_button_press_event_cb(event_box_ptr::Ptr,eventptr::Ptr, tab)
+    event_box = convert(GtkEventBox, event_box_ptr)
+    event     = convert(Gtk.GdkEvent,eventptr)
+    if (event.button == 3)
```

Comment:
Replace by this

    mod = get_default_mod_mask()
    if event.button == 3 || (
        event.button == 1 && event.state & mod == SecondaryModifer
    )

That makes it work on macbook (you have to control click to right-click)